.test-infra/jenkins/LoadTestConfig.groovy

```diff
@@ -0,0 +1,621 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import CommonTestProperties.Runner
+import CommonTestProperties.SDK
+import LoadTestConfig.SerializableOption
+import groovy.json.JsonBuilder
+import org.codehaus.groovy.runtime.InvokerHelper
+
+import java.util.function.Predicate
+
+import static java.util.Objects.nonNull
+import static java.util.Objects.requireNonNull
+
+/**
+ * This class contains simple DSL for load tests configuration. Configuration as Map<String, Serializable>
+ * [{@link LoadTestConfig#config config} -- returns configuration map]
+ * [{@link LoadTestConfig#templateConfig templateConfig} -- return LoadTestConfig reusable object]
+ * [{@link LoadTestConfig#fromTemplate fromTemplate} -- returns configuration from given template].<br><br>
+ *
+ * Example:
+ * <blockquote><pre>
+ * LoadTestConfig template = templateConfig {
+ *     title 'Load test'
+ *     test 'Beam Load test'
+ *     dataflow()
+ *     pipelineOptions {
+ *         python()
+ *         jobName 'Any job name'
+ *         publishToBigQuery true
+ *         //other fields
+ *     }
+ *     specificParameters([
+ *          fanout: 4
+ *     ])
+ * }
+ * Map<String, Serializable> configMap = fromTemplate(template) {
+ *     //fields can be changed or/and added
+ *     flink()
+ *     pipelineOptions {
+ *         parallelism 5
+ *         inputOptions {
+ *             numRecords 20000
+ *             keySize 1000
+ *             valueSize 10
+ *         }
+ *     }
+ * }
+ * </pre></blockquote>
+ */
+class LoadTestConfig implements SerializableOption<Map<String, Serializable>> {
+
+    private String _title
+    private String _test
+    private Runner _runner
+    private PipelineOptions _pipelineOptions
+
+    private LoadTestConfig() {}
+
+    void title(final String title) { _title = title }
+    void test(final String test) { _test = test }
+
+    //runners
+    void dataflow() { setRunnerAndUpdatePipelineOptions(Runner.DATAFLOW)}
+    void spark() { setRunnerAndUpdatePipelineOptions(Runner.SPARK) }
```

Comment:
TBH I looked at `CommonTestProperties` class and picked `Runner` values. Since I am not that much familiar with code yet, my implementation (the one above) can be wrong. Should I limit it to dataflow and portable? (at least for the time being)